From patches Mon Sep 17 00:00:00 2001
From: Kazuya Saito <saito.kazuya@jp.fujitsu.com>
To: "qemu-devel@nongnu.org" <qemu-devel@nongnu.org>
Cc: "kvm@vger.kernel.org" <kvm@vger.kernel.org>,
 Paolo Bonzini <pbonzini@redhat.com>,
 "aliguori@us.ibm.com" <aliguori@us.ibm.com>,
 "stefanha@linux.vnet.ibm.com" <stefanha@linux.vnet.ibm.com>
Date: Fri, 29 Mar 2013 13:27:52 +0900
Message-id: 51551848.3040207@jp.fujitsu.com
Subject: [PATCH 2/2] kvm-all: add kvm_run_exit tracepoint

This patch enable us to know exit reason of KVM_RUN. It will help us
know where the trouble is caused.

Signed-off-by: Kazuya Saito <saito.kazuya@jp.fujitsu.com>
---
 kvm-all.c    |    1 +
 trace-events |    2 ++
 2 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/kvm-all.c b/kvm-all.c
index fdb099c..325f5e7 100644
--- a/kvm-all.c
+++ b/kvm-all.c
@@ -1574,6 +1574,7 @@ int kvm_cpu_exec(CPUArchState *env)
             abort();
         }

+        trace_kvm_run_exit(cpu->cpu_index, run->exit_reason);
         switch (run->exit_reason) {
         case KVM_EXIT_IO:
             DPRINTF("handle_io\n");
diff --git a/trace-events b/trace-events
index 3023744..8fd6e80 100644
--- a/trace-events
+++ b/trace-events
@@ -1106,3 +1106,5 @@ migrate_set_state(int new_state) "new state %d"
 kvm_ioctl(int type, void *arg) "type %d, arg %p"
 kvm_vm_ioctl(int type, void *arg) "type %d, arg %p"
 kvm_vcpu_ioctl(int cpu_index, int type, void *arg) "cpu_index %d, type %d, arg %p"
+kvm_run_exit(int cpu_index, uint32_t reason) "cpu_index %d, reason %d"
+
-- 
1.7.1




